lms/static/js/learner_dashboard/views/course_entitlement_view.js

```diff
@@ -0,0 +1,356 @@
+                                 ' lose any progress you have made in this session.')
+                     } else {
+                         confirmationMsgTitle = gettext('Are you sure that you would like to leave this session?');
+                         confirmationMsgBody = gettext('Please know that by leaving you will lose any progress you ' +
+                                'had made in this session.')
+                     }
+
+                     // Remove existing popover and re-initialize
+                     popoverDialogHtml = this.verificationTpl({
+                         confirmationMsgTitle: confirmationMsgTitle,
+                         confirmationMsgBody: confirmationMsgBody
+                     });
+
+                     this.$('.enroll-btn-initial').popover({
+                         placement: 'bottom',
+                         container: this.$el,
+                         html: true,
+                         trigger: 'click',
+                         content: popoverDialogHtml.text
+                     });
+                 },
+
+                 removeVerificationDialog: function() {
+                     /* Removes the Bootstrap v4 dialog modal from the update session enrollment button. */
+                     this.$('.enroll-btn-initial').popover('dispose');
+                 },
+
+                 hideVerificationDialog: function() {
+                     /* Hides the Bootstrap v4 dialog modal without removing it from the DOM. */
+                     this.$('.enroll-btn-initial').focus().popover('hide');
+                 },
+
+                 handleVerificationPopoverA11y: function(e) {
+                     /* Ensure that the second step verification popover is treated as an a11y compliant dialog */
+                     var nextButton,
+                         openButton = $(e.target).closest('.course-entitlement-selection-container')
+                             .find('.enroll-btn-initial');
+                     if (e.key === 'Tab') {
+                         e.preventDefault();
+                         nextButton = $(e.target).is(':first-child') ? $(e.target).next('.final-confirmation-btn') :
+                             $(e.target).prev('.final-confirmation-btn');
+                         nextButton.focus();
+                     } else if (e.key === 'Escape') {
+                         openButton.popover('hide');
+                         openButton.focus();
+                     }
+                 },
+
+                 formatCourseHomeUrl: function(sessionKey) {
+                     /*
+                     Takes the base course home URL and updates it with the new session id, leveraging the
+                     the fact that all course keys contain a '+' symbol.
+                     */
+                     var oldSessionKey = this.courseHomeUrl.split('/')
+                         .filter(
+                             function(x) {
```

Comment:
Please update to replace ```x``` with a more meaningful variable name to make the code easier to read.

Reply:
Changed